src/main/java/nl/mpcjanssen/simpletask/task/TodoList.kt

```diff
@@ -304,15 +309,11 @@ object TodoList {
         todoItemsDao.updateInTx(items)
     }
 
-    fun archive(todoFilename: String, doneFileName: String, tasks: List<TodoItem>?, eol: String) {
+    fun archive(todoFilename: String, doneFileName: String, tasks: List<TodoItem>, eol: String) {
         ActionQueue.add("Archive", Runnable {
-            val items = todoItems
-            val tasksToArchive = tasks ?: items
-
-            val completedTasks = tasksToArchive.filter { it.task.isCompleted() }
             try {
-                FileStore.appendTaskToFile(doneFileName, completedTasks.map { it.task.text }, eol)
-                todoItemsDao.deleteInTx(completedTasks)
+                FileStore.appendTaskToFile(doneFileName, tasks.map { it.task.text }, eol)
+                todoItemsDao.deleteInTx(tasks)
```

Comment:
We're always explicitly passing in the tasks we want to archive, so we don't need to build a list ourselves.

Inadvertent side effect: almost fixes #496